--- /home/mockbuild/svn_repos/toolkit-building/kickstarts/centos6-base.cfg	2015-06-18 11:59:07.964168524 -0400
+++ /home/mockbuild/svn_repos/toolkit-building/kickstarts/centos6-fullinstall.cfg	2015-06-18 11:55:24.902247646 -0400
@@ -1,5 +1,4 @@
 install
-url --url=http://linux.mirrors.es.net/centos/6/os/i386
 
 ##############################
 # System Configuration
@@ -22,8 +21,6 @@
 zerombr
 bootloader --location=mbr
 
-#clearpart --all
-
 network --onboot=yes
 
 ##############################
@@ -33,16 +30,6 @@
 text
 
 ##############################
-# Repository Configuration
-##############################
-repo --name=a-base      --baseurl=http://linux.mirrors.es.net/centos/6/os/i386 --excludepkgs=kernel*
-repo --name=a-updates   --mirrorlist=http://software.internet2.edu/rpms/el6/mirrors-Toolkit-CentOS-Updates-i386 --excludepkgs=kernel*
-repo --name=a-EPEL      --mirrorlist=http://software.internet2.edu/rpms/el6/mirrors-Toolkit-EPEL-i386
-repo --name=a-Internet2 --baseurl=http://software.internet2.edu/rpms/el6/i386/main/
-repo --name=a-Web100    --baseurl=http://software.internet2.edu/web100_kernel/rpms/el6/i386/main/
-repo --name=a-Internet2-staging	--baseurl=http://software.internet2.edu/branches/staging/rpms/el6/i386/main/
-
-##############################
 # Install Packages
 ##############################
 %packages
@@ -50,6 +37,10 @@
 @core
 @console-internet
 
+# Anaconda is needed for revisor to build the DVD
+anaconda
+syslinux
+
 authconfig
 bash
 binutils
