--- /home/alake/svn_repos/perfsonar-ps/Toolkit_Building/kickstarts/centos6-base.cfg	2014-10-22 07:09:44.408900666 -0700
+++ /home/alake/svn_repos/perfsonar-ps/Toolkit_Building/kickstarts/centos6-fullinstall.cfg	2014-10-22 07:09:44.409900635 -0700
@@ -1,5 +1,4 @@
 install
-url --url=http://linux.mirrors.es.net/centos/6/os/i386
 
 ##############################
 # System Configuration
@@ -22,8 +21,6 @@
 zerombr
 bootloader --location=mbr
 
-#clearpart --all
-
 network --onboot=yes
 
 ##############################
@@ -33,15 +30,6 @@
 text
 
 ##############################
-# Repository Configuration
-##############################
-repo --name=a-base      --baseurl=http://linux.mirrors.es.net/centos/6/os/i386 --excludepkgs=kernel*
-repo --name=a-updates   --mirrorlist=http://software.internet2.edu/branches/epel6-repo/rpms/mirrors-Toolkit-CentOS-Updates-i386 --excludepkgs=kernel*
-repo --name=a-EPEL      --mirrorlist=http://software.internet2.edu/branches/epel6-repo/rpms/mirrors-Toolkit-EPEL-i386
-repo --name=a-Internet2 --baseurl=http://software.internet2.edu/rpms/el6/i386/main/
-repo --name=a-Web100    --baseurl=http://software.internet2.edu/web100_kernel/rpms/el6/i386/main/
-
-##############################
 # Install Packages
 ##############################
 %packages
@@ -49,6 +37,10 @@
 @core
 @console-internet
 
+# Anaconda is needed for revisor to build the DVD
+anaconda
+syslinux
+
 authconfig
 bash
 binutils
