--- /home/mockbuild/svn_repos/perfsonar-ps/Toolkit_Building/kickstarts/centos6-base.cfg	2014-08-06 18:55:04.631024574 -0400
+++ /home/mockbuild/svn_repos/perfsonar-ps/Toolkit_Building/kickstarts/centos6-liveusb.cfg	2014-08-06 18:55:04.631024574 -0400
@@ -1,5 +1,5 @@
 install
-url --url=http://linux.mirrors.es.net/centos/6/os/i386
+url --url=http://linux.mirrors.es.net/centos/6/os/[BUILD_ARCH]
 
 ##############################
 # System Configuration
@@ -22,9 +22,12 @@
 zerombr
 bootloader --location=mbr
 
-#clearpart --all
-
-network --onboot=yes
+# Specify network boot devices for LiveCD
+network --bootproto dhcp --device eth0 --onboot=yes
+network --bootproto dhcp --device eth1 --onboot=yes
+network --bootproto dhcp --device eth2 --onboot=yes
+network --bootproto dhcp --device eth3 --onboot=yes
+network --bootproto dhcp --device eth4 --onboot=yes
 
 ##############################
 # Installation Configuration
@@ -35,11 +38,11 @@
 ##############################
 # Repository Configuration
 ##############################
-repo --name=a-base      --baseurl=http://linux.mirrors.es.net/centos/6/os/i386 --excludepkgs=kernel*
-repo --name=a-updates   --mirrorlist=http://software.internet2.edu/branches/epel6-repo/rpms/mirrors-Toolkit-CentOS-Updates-i386 --excludepkgs=kernel*
-repo --name=a-EPEL      --mirrorlist=http://software.internet2.edu/branches/epel6-repo/rpms/mirrors-Toolkit-EPEL-i386
-repo --name=a-Internet2 --baseurl=http://software.internet2.edu/rpms/el6/i386/main/
-repo --name=a-Web100    --baseurl=http://software.internet2.edu/web100_kernel/rpms/el6/i386/main/
+repo --name=a-base      --baseurl=http://linux.mirrors.es.net/centos/6/os/[BUILD_ARCH] --excludepkgs=kernel*
+repo --name=a-updates   --mirrorlist=http://software.internet2.edu/rpms/el6/mirrors-Toolkit-CentOS-Updates-[BUILD_ARCH] --excludepkgs=kernel*
+repo --name=a-EPEL      --mirrorlist=http://software.internet2.edu/rpms/el6/mirrors-Toolkit-EPEL-[BUILD_ARCH]
+repo --name=a-Internet2 --baseurl=http://software.internet2.edu/rpms/el6/[BUILD_ARCH]/main/
+repo --name=a-Web100    --baseurl=http://software.internet2.edu/web100_kernel/rpms/el6/[BUILD_ARCH]/main/
 repo --name=a-DataStax  --baseurl=http://rpm.datastax.com/community
 
 ##############################
@@ -129,6 +132,7 @@
 owamp-server
 
 perl-perfSONAR_PS-Toolkit
+perl-perfSONAR_PS-Toolkit-LiveCD
 perl-perfSONAR_PS-Toolkit-SystemEnvironment
 
 perl-perfSONAR_PS-LSCacheDaemon
@@ -148,7 +152,18 @@
 ##############################
 # Run Post Scripts
 ##############################
-%post --log=/root/post_install.log
+%post --nochroot --log=$INSTALL_ROOT/root/post_install.log
+
+cp /etc/resolv.conf $INSTALL_ROOT/etc/resolv.conf
+/usr/sbin/chroot $INSTALL_ROOT <<EOS
+
+##############################
+# Make sure some permissions are correct
+##############################
+chown mysql:mysql /var/run/mysqld
+chown perfsonar:perfsonar /opt/SimpleLS/bootstrap/etc/*
+chown perfsonar:perfsonar /opt/perfsonar_ps/*/etc/*
+chown apache:apache /opt/perfsonar_ps/serviceTest/etc/*
 
 ##############################
 # Initialize the MySQL Databases
@@ -158,7 +173,41 @@
 /sbin/service mysqld stop
 
 ##############################
+# Add the LiveUSB keyword
+##############################
+echo "site_project=pS-NPToolkit-LiveUSB" >> /opt/perfsonar_ps/toolkit/etc/administrative_info
+echo "site_project=pS-NPToolkit-LiveUSB" >> /opt/perfsonar_ps/ls_registration_daemon/etc/ls_registration_daemon.conf
+
+##############################
 # Disable Readahead
 ##############################
 sed -i 's/=\"yes\"/=\"no\"/g' /etc/sysconfig/readahead
+ 
+##############################
+# Remove Un-needed Cron Scripts
+##############################
+rm -f /etc/cron.daily/cups
+rm -f /etc/cron.daily/makewhatis.cron
+rm -f /etc/cron.daily/mlocate.cron
+rm -f /etc/cron.daily/prelink
+rm -f /etc/cron.daily/rpm
+rm -f /etc/cron.weekly/makewhatis.cron
+
+##############################
+# Force Enable Shadow Passwords
+##############################
+/usr/sbin/pwconv
+
+##############################
+# Force Root Password Change
+##############################
+/usr/bin/chage -d 0 root
+EOS
+
+##############################
+# Cleanup
+##############################
+rm $INSTALL_ROOT/root/.bash_history
+rm $INSTALL_ROOT/etc/resolv.conf
+touch $INSTALL_ROOT/etc/resolv.conf
 %end
