--- aufs-cvs-20080414.ori/fs/aufs/cpup.h	2008-04-14 01:38:24.000000000 +0200
+++ aufs-cvs-20080414/fs/aufs/cpup.h	2008-04-14 15:21:37.000000000 +0200
@@ -34,7 +34,7 @@
 static inline
 void au_cpup_attr_blksize(struct inode *inode, struct inode *h_inode)
 {
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 18)
 	inode->i_blksize = h_inode->i_blksize;
 #endif
 }
